charts/Makefile.common

```
@@ -63,11 +63,26 @@ _create-namespace:
 
 .PHONY: _delete-namespace
 _delete-namespace:
-	$(KUBECTL_BIN) delete namespace $(NAMESPACE) --ignore-not-found=true
+	@echo "Deleting namespace $(NAMESPACE)..."
+
+	$(KUBECTL_BIN) delete namespace $(NAMESPACE) --wait --timeout=60s --ignore-not-found=true; \
+	echo "Waiting for namespace $(NAMESPACE) to terminate..."; \
+	for i in {1..30}; do \
+		if ! $(KUBECTL_BIN) get namespace $(NAMESPACE) >/dev/null 2>&1; then \
+			echo "Namespace $(NAMESPACE) deleted successfully."; \
+			exit 0; \
+		fi; \
+		sleep 2; \
+	done; \
+
+	echo "Timeout waiting for namespace $(NAMESPACE) to terminate. Proceeding anyway."; \
+	exit 0
+
+	@echo "Done deleting namespace $(NAMESPACE)"
 
 .PHONY: _delete-crds
 _delete-crds:
-	$(KUBECTL_BIN) get crds -o name | grep '$(CRD_NAME)' | xargs $(KUBECTL_BIN) delete --ignore-not-found=true || true
+	$(KUBECTL_BIN) get crds -o name | grep '$(CRD_NAME)' | xargs -r -n1 timeout 30s $(KUBECTL_BIN) delete --ignore-not-found=true || true	
```

I still don't understand how this works

---

@emoreth `timeout` is wrapping the `kubectl` delete command, and the `-n1` ensures that each CRD is processed one at a time.

Without `-n1`, `xargs` could pass multiple CRDs to a single `kubectl` delete, and if any one of them hangs, the whole command gets stuck which is what I was running into. Adding `-n1` helps isolate each delete operation and makes sure the timeout applies per CRD.

---

?

Why using timeout on kubectl and ignoring errors?

---

makes sense @emoreth, removed `-n1` and timeout. Now it works without them as well